site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: *.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: gateway-b
+spec:
+  gatewayClassName: nadir-lb
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: subdomain.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-parent
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  - name: gateway-b
+    sectionName: http
+  rules:
+    - allowedRoutes:
+        kinds: 
+          - kind: HTTPRoute
+        namespaces:
+          from: Same
+    - backendRefs:
+      - name: default-svc
+        port: 8080
+status:
+  parents:
+  - controllerName: acme
+    parentRef:
+      name: gateway-a
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Successfully accepted
+  - controllerName: nadir
+    parentRef:
+      name: gateway-b
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Successfully accepted
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-child
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  hostnames:
+    - testdomain.example.com
+  rules:
+    - backendRefs:
+      - name: foo-svc
+        port: 8080
+status:
+  parents:
+  - controllerName: acme
+    parentRef:
+      name: http-parent
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Accepted successfully
+    grandparents:
+      - parentRef:
+          name: gateway-a
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          message: Successfully accepted
+  - controllerName: nadir
+    parentRef:
+      name: http-parent
+    conditions:
+    - type: Accepted
+      status: true
+      reason: GrandparentConfigError
+      message: Some configuration was not accepted in the attachment process. See grandparents for more details.
+    grandparents:
+      - parentRef:
+          name: gateway-b
+        conditions:
+        - type: Accepted
+          status: false
+          reason: HostnameMismatch
+          message: The route could not be accepted because of a hostname mismatch - `testdomain.example.com` does not match `subdomain.example.com`.
+```
+
+
+
+## Alternatives
+
+### Extending backendRef to allow targeting a Route
+
+This idea was to allow `backendRef` to, optionally, specify a route instead of a
+service. Some of the reasons for not taking this approach include:
+
+- It would require changes to `backendRefs`. No changes to `allowedRoutes` are
+  needed to use it for this. 
+- To minimize the changes to `backendRef`, it would require the owner of the
+  parent route to know the exact name of the child route. If the two routes are
+  in different namespaces, that might require cross-organization coordination
```

Comment:
This is already needed though, just in one direction.

Reply:
I think it's much more reasonable to require a child to know details of the parent resource than the other way around. As an owner of a parent route, I don't want to have to worry about what Route(s) are named in namespace foo, just that I trust Route owners in that namespace to handle requests under `/foo`.

Reply:
I think an important use case is to allow helm charts to be written and independent of cluster config.

For example in servlets web.xml routes are independent of hostname ( and even context prefix ). This is a very useful property and allows a .war to be deployed in any server without changes. 

Not sure if 'parent' knowing the naming details in the child namespace is true - the BackendRef has a name and namespace, and the name is quite significant and required and must be the exact Service name. Why would delegating
to a route be different ?